xarray/tests/test_datatree.py

```diff
@@ -494,11 +494,11 @@ def test_full(self, simple_datatree):
         assert paths == [
             "/",
             "/set1",
+            "/set2",
+            "/set3",
```

Comment:
Nice. Interesting that this is the only test that explicitly checks the order of iteration.